models/dataset.py

```diff
@@ -0,0 +1,157 @@
+"""Open datasets and processes them to be used by a neural network."""
+
+
+import json
+import os
+import numpy as np
+import h5py
+import torch
+from torch.utils.data import DataLoader, random_split
+import functools
+from PIL import Image
+
+CUDA = torch.cuda.is_available()
+
+
+KWARGS = {'num_workers': 1, 'pin_memory': True} if CUDA else {}
+
+
+def open_dataset(path, new_size, is_3d):
+    """
+    Open datasets and processes data in ordor to make tensors.
+    Parameters
+    ----------
+    path : string path.
+    new_size : int.
+    is_3d : boolean if 2d or 3d
+    Returns
+    -------
+    tensor, images in black and white.
+    """
+    if not os.path.exists(path):
+        raise OSError
+    if path.lower().endswith(".h5"):
+        data_dict = h5py.File(path, 'r')
+        all_datasets = data_dict['particles'][:]
+    else:
+        all_datasets = np.load(path)
+    dataset = np.asarray(all_datasets)
+    img_shape = dataset.shape
+    n_imgs = img_shape[0]
+    new_dataset = []
+    if is_3d:
+        dataset = torch.Tensor(dataset)
+        dataset = normalization_linear(dataset)
+    else:
+        if img_shape.ndim == 3:
+            for i in range(n_imgs):
+                new_dataset.append(np.asarray(Image.fromarray(
+                    dataset[i]).resize([new_size, new_size])))
+        elif img_shape.ndim == 4:
+            for i in range(n_imgs):
+                new_dataset.append(np.asarray(Image.fromarray(
+                    dataset[i][0]).resize([new_size, new_size])))
+        dataset = torch.Tensor(new_dataset)
+        dataset = normalization_linear(dataset)
+        if len(img_shape) != 4:
+            dataset = dataset.reshape(
+                (img_shape[0], 1, img_shape[1], img_shape[1]))
+    return dataset
+
+
+def normalization_linear(dataset):
+    """
+    Normalize a tensor.
+    Parameters
+    ----------
+    dataset : tensor, images.
+    Returns
+    -------
+    dataset : tensor, normalized images.
+    """
+    for i, data in enumerate(dataset):
+        min_data = torch.min(data)
+        max_data = torch.max(data)
+        if max_data == min_data:
+            raise ZeroDivisionError
+        dataset[i] = (data - min_data) / (max_data - min_data)
+    return dataset
+
+
+def organize_dataset(dataset, batch_size, frac_val):
```

Comment:
organize_dataset -> split_dataset

(naming is important, our future users will heavily rely on it to understand what the functions do).